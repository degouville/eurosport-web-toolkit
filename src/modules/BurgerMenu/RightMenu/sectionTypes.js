// right menu section column/menu types;

export const COLUMN = {
  POPULAR_SPORTS: 'popular-sports',
  ALL_SPORTS: 'all-sports',
  ABOUT_US: 'about-us',
};

export const MENU = {
  SPORTS: 'sports',
};
