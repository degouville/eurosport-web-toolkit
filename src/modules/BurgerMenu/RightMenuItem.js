/* eslint-disable react/prop-types */
import React from 'react';
import styled, { css } from 'react-emotion';
import kebabCase from 'lodash/kebabCase';
import { large } from '../../breakpoints';
import { arsenic, cerulean, regentGray, turquoiseBlue, whiteLilac } from '../../colors';

const StyledItem = styled.li`
  list-style-type: none;
  padding: 0;
  margin: 0;
  line-height: 50px;

  ${large(css`
    line-height: 55px;
  `)};
`;

const StyledItemLink = styled.a`
  color: ${arsenic};
  font-weight: bold;
  text-decoration: none;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 20px;

  ${large(css`
    font-size: 24px;
  `)};

  &:hover {
    color: ${regentGray};
  }

  ${props =>
    props.columnType === 'popular-sports' &&
    css`
      color: ${cerulean};
      &:hover {
        color: ${turquoiseBlue};
      }
    `};

  ${props =>
    props.columnType === 'all-sports' &&
    props.item.items.length > 0 &&
    css`
      display: flex;
      justify-content: space-between;
      &::after {
        content: '>';
        color: ${whiteLilac};
      }
      ${large(css`
        &::after {
          display: none;
        }
      `)};
    `};
`;

const RightMenuItem = ({ item, menuType, columnType, isMobileMenu, onSubMenuOpen }) => (
  <StyledItem>
    <StyledItemLink
      onClick={e => {
        if (menuType === 'sports' && isMobileMenu && item.items.length) {
          e.preventDefault();
          onSubMenuOpen(item);
        }
      }}
      href={item.link ? item.link.url : '#'}
      item={item}
      columnType={columnType}
      data-test={kebabCase(
        `${menuType === 'sports' && isMobileMenu && item.items.length ? 'mobile' : 'desktop'}-${item.name}`
      )}
    >
      {item.name}
    </StyledItemLink>
  </StyledItem>
);

export default RightMenuItem;
