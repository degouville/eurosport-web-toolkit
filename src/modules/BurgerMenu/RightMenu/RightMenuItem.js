/* eslint-disable react/prop-types */
import React from 'react';
import styled, { css } from 'react-emotion';
import kebabCase from 'lodash/kebabCase';
import { medium, large } from '../../../breakpoints';
import { arsenic, cerulean, regentGray, turquoiseBlue, whiteLilac } from '../../../colors';
import * as types from './sectionTypes';

const StyledItem = styled.li`
  list-style-type: none;
  padding: 0;
  margin: 0;
  line-height: 50px;
`;

const StyledItemLink = styled.a`
  color: ${arsenic};
  font-weight: bold;
  text-decoration: none;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 16px;

  ${medium(css`
    font-size: 18px;
  `)};

  ${large(css`
    font-size: 24px;
  `)}

  &:hover {
    color: ${regentGray};
  }

  ${props =>
    props.columnType === types.COLUMN.POPULAR_SPORTS &&
    css`
      color: ${cerulean};
      &:hover {
        color: ${turquoiseBlue};
      }
    `};

  ${({ columnType, item }) =>
    columnType === types.COLUMN.ALL_SPORTS &&
    item.items &&
    item.items.length > 0 &&
    css`
      display: flex;
      justify-content: space-between;
      &::after {
        content: '>';
        color: ${whiteLilac};
        margin-right: 10px;
      }
      ${large(css`
        &::after {
          display: none;
        }
      `)};
    `};
`;

const RightMenuItem = ({ item, menuType, columnType, isMobileMenu, onSubMenuOpen }) => (
  <StyledItem>
    <StyledItemLink
      onClick={e => {
        if (menuType === types.MENU.SPORTS && isMobileMenu && item.items && item.items.length) {
          e.preventDefault();
          onSubMenuOpen(item);
        }
      }}
      href={item.link ? item.link.url : '#'}
      item={item}
      columnType={columnType}
      data-test={kebabCase(
        `${menuType === types.MENU.SPORTS && isMobileMenu && item.items && item.items.length ? 'mobile' : 'desktop'}-${
          item.name
        }`
      )}
    >
      {item.name}
    </StyledItemLink>
  </StyledItem>
);

export default RightMenuItem;
