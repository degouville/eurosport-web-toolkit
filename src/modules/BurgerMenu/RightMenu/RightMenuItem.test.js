import React from 'react';
import { shallow } from 'enzyme';
import RightMenuItem, { StyledItemLink } from './RightMenuItem';
import * as types from './sectionTypes';

describe('RightMenuItem', () => {
  describe('Snapshot testing', () => {
    it('renders with expected snapshot for mobile', () => {
      expect(
        shallow(
          <RightMenuItem
            isMobileMenu
            item={{ id: 1, name: 'foo', sections: [] }}
            key={1}
            onMenuSelected={jest.fn()}
            selectedMenuId={2}
          />
        )
      ).toMatchSnapshot();
    });

    it('renders with expected snapshot for desktop', () => {
      expect(
        shallow(
          <RightMenuItem
            isMobileMenu={false}
            item={{ id: 1, name: 'foo', sections: [] }}
            key={1}
            onMenuSelected={jest.fn()}
            selectedMenuId={2}
          />
        )
      ).toMatchSnapshot();
    });
  });

  it('should call onMenuSelected callback when clicked', () => {
    // Given
    const onSubMenuOpenCallback = jest.fn();
    const wrapper = shallow(
      <RightMenuItem
        isMobileMenu
        item={{ id: 1, items: [{}], name: 'foo', sections: [] }}
        key={1}
        menuType={types.MENU.SPORTS}
        onSubMenuOpen={onSubMenuOpenCallback}
        selectedMenuId={2}
      />
    );
    const styledItemLink = wrapper.find(StyledItemLink);

    // When
    styledItemLink.simulate('click', new Event('click'));

    // Expect
    expect(onSubMenuOpenCallback).toHaveBeenCalled();
  });
});
