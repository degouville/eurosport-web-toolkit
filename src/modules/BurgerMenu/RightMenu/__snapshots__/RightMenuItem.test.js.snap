// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightMenuItem Snapshot testing renders with expected snapshot for desktop 1`] = `
<Styled(li)>
  <Styled(a)
    data-test="desktop-foo"
    href="#"
    item={
      Object {
        "id": 1,
        "name": "foo",
        "sections": Array [],
      }
    }
    onClick={[Function]}
  >
    foo
  </Styled(a)>
</Styled(li)>
`;

exports[`RightMenuItem Snapshot testing renders with expected snapshot for mobile 1`] = `
<Styled(li)>
  <Styled(a)
    data-test="desktop-foo"
    href="#"
    item={
      Object {
        "id": 1,
        "name": "foo",
        "sections": Array [],
      }
    }
    onClick={[Function]}
  >
    foo
  </Styled(a)>
</Styled(li)>
`;
